src/dsl.jl

```diff
@@ -286,112 +277,88 @@ end
 ### DSL Internal Master Function ###
 
 # Function for creating a ReactionSystem structure (used by the @reaction_network macro).
-function make_reaction_system(ex::Expr; name = :(gensym(:ReactionSystem)))
+function make_reaction_system(ex::Expr, name)
 
-    # Handle interpolation of variables
+    # Handle interpolation of variables in the input.
     ex = esc_dollars!(ex)
 
-    # Read lines with reactions and options.
+    # Extracts the lines with reactions and lines with options.
     reaction_lines = Expr[x for x in ex.args if x.head == :tuple]
     option_lines = Expr[x for x in ex.args if x.head == :macrocall]
 
-    # Get macro options.
-    if length(unique(arg.args[1] for arg in option_lines)) < length(option_lines)
+    # Extracts the options used (throwing errors for repeated options).
+    if !allunique(arg.args[1] for arg in option_lines)
         error("Some options where given multiple times.")
     end
-    options = Dict(map(arg -> Symbol(String(arg.args[1])[2:end]) => arg,
-        option_lines))
-
-    # Reads options.
-    default_reaction_metadata = :([])
-    check_default_noise_scaling!(default_reaction_metadata, options)
-    compound_expr, compound_species = read_compound_options(options)
-    continuous_events_expr = read_events_option(options, :continuous_events)
-    discrete_events_expr = read_events_option(options, :discrete_events)
+    options = Dict(Symbol(String(arg.args[1])[2:end]) => arg for arg in option_lines)
 
-    # Parses reactions, species, and parameters.
-    reactions = get_reactions(reaction_lines)
+    # Reads options (round 1, options which must be read before the reactions, e.g. because 
+    # they might declare parameters/species/variables).
+    compound_expr_init, compound_species = read_compound_options(options)
     species_declared = [extract_syms(options, :species); compound_species]
     parameters_declared = extract_syms(options, :parameters)
     variables_declared = extract_syms(options, :variables)
+    vars_extracted, add_default_diff, equations = read_equations_options(options, 
+        variables_declared)
 
-    # Reads more options.
-    vars_extracted, add_default_diff, equations = read_equations_options(
-        options, variables_declared)
+    # Extracts all reactions. Extracts all parameters, species, and variables of the system and
+    # creates lists with them.
+    reactions = get_reactions(reaction_lines)
     variables = vcat(variables_declared, vars_extracted)
-
-    # handle independent variables
```

Comment:
Part of the organisations so that options are handled together (in two sets, some before and some after the reactions are considered, depending on whichever is possible for which option)